src/App.jsx

```diff
@@ -15,41 +11,57 @@ class App extends React.Component {
       actor: '',
       permission: '',
       session: '',
-      accountData: {}
+      accountData: {},
+      error: '',
     };
   }
 
-  componentDidMount = async ()  => { 
+  componentDidMount = async () => {
     this.checkIfLoggedIn();
-  }
+  };
 
   checkIfLoggedIn = async () => {
-    const { auth, accountData } = await ProtonSDK.restoreSession();
-    if (auth && auth.actor && auth.permission) {
-      this.setLoggedInState(auth.actor, auth.permission, accountData);
+    const { auth, accountData, error } = await ProtonSDK.restoreSession();
+    if (error || !(auth && auth.actor && auth.permission)) {
+      this.setErrorState(error || "Restore error");
```

Comment:
Let's update the message to be a little more user friendly, maybe something like "Unable to restore previous session."